File: Week10/H-Index/hyewon.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

int solution(vector<int> citations) {
	int answer = 0;
	sort(citations.begin(), citations.end());

	for (int i = 0; i < citations.size(); i++) {
		int cnt = citations[i];
		int bigger = 0;
		for (int j = 0; j < citations.size(); j++) {
			if (citations[j] >= cnt)
				bigger++; //h번 이상 인용된 논문의 개수
		}	
		int smaller = citations.size() - bigger;
		if (bigger >= cnt && smaller < cnt) {
			if (answer < cnt)
				answer = cnt;
		}
	}
	return answer;
}

int main() {
	cout<< solution({ 3,0,6,1,5 });
}

File: Week11/PhoneBook/hyewon.py
def solution(phone_book):
    phone_book.sort()
    answer = True
    for i in range(len(phone_book)):
        for j in range(i+1, len(phone_book)):
            if phone_book[j].startswith(phone_book[i]):
                return False
    return answer

print(solution(["119", "97674223", "1195524421"]))
print(solution(["123","456","789"]))
print(solution(["12","123","1235","567","88"]))